### PR TITLE
added py sol for biparteness cf 862 b

### PR DESCRIPTION
**Pull request by alexyd88**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Review by ryanchou-dev — COMMENTED** [review]
this looks good -- if possible id prefer u do it recursively (does increasing the maximum recursion depth work?)

if not, ill merge on next approval.

**Comment by alexyd88** [comment]
> this looks good -- if possible id prefer u do it recursively (does increasing the maximum recursion depth work?)
> 
> if not, ill merge on next approval.

yeah, i tried it but it doesn't work. should I add a comment talking about increasing recursion depth?

**Comment by alexyd88** [comment]
> this looks good -- if possible id prefer u do it recursively (does increasing the maximum recursion depth work?)
> 
> if not, ill merge on next approval.

yeah, i tried it but it doesn't work. should I add a comment talking about increasing recursion depth?

File: solutions/silver/cf-862B.mdx [hunk 1]
```diff
@@ -115,5 +115,48 @@ int main() {
 }
 ```
 </CPPSection>
+<PySection>
 
+```py
+MAX_N = 100000
+
+# +1 for 1-indexed nodes
+adj = [[] for _ in range(MAX_N + 1)]
```
Comment (SansPapyrus683) [inline solutions/silver/cf-862B.mdx:124]: can't you size this dynamically?

Comment (alexyd88) [reply solutions/silver/cf-862B.mdx:124]: For adjacency list I thought it was standard to use for _ in range. How would that work dynamically? I think just making it adj = [] will give RE.

Comment (SansPapyrus683) [reply solutions/silver/cf-862B.mdx:124]: like define it based off `n`

Comment (alexyd88) [reply solutions/silver/cf-862B.mdx:124]: oh yeah, you're right

File: solutions/silver/cf-862B.mdx [hunk 2]
```diff
@@ -115,5 +115,42 @@ int main() {
...
-
+<PySection>
+
+```py
+# counters for colored and uncolored nodes
+c = 0
+uc = 0
+
+
+def dfs(node, parent, col):
```
Comment (SansPapyrus683) [inline solutions/silver/cf-862B.mdx:126]: type annotations

**Review by SansPapyrus683 — APPROVED** [review]
just one thing

**Comment by SansPapyrus683** [comment]
> > this looks good -- if possible id prefer u do it recursively (does increasing the maximum recursion depth work?)
> > if not, ill merge on next approval.
> 
> yeah, i tried it but it doesn't work. should I add a comment talking about increasing recursion depth?

add a comment and see how it works

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**